torch/distributed/_tensor/api.py

```diff
@@ -418,6 +418,9 @@ def to_local(
         .. note:: `to_local` is differentiable, the `requires_grad` of the local tensor returned
             will depend on if the `DTensor` requires_grad or not.
         """
+        if not torch.is_grad_enabled():
```

Comment:
Now the question is, how expensive is `torch.is_grad_enabled()` 😆

Reply:
I hope it's pretty cheap given that we are using it invasively in pytorch ;)

Reply:
tested and got a trace locally, I didn't even see the `to_local` or is_grad_enabled calls in the CPU trace :)

Reply:
I guess that's because there is no aten op in that call stack, so the profiler cannot see it? (but this could still contribute to gaps between other ops in the profiler trace?)